build.cake

```diff
@@ -0,0 +1,108 @@
+var sourceDir = "./exercises/";
+var projectDirs = GetFiles(buildDir + "**/" + project + ".csproj");
+
+var allProjects = GetFiles(buildDir + "**/*.csproj");
+
+/** The following exercises provide a stub implementation and the 
+    user has to complete the implementation. Therefore compiling 
+    these will result in failed compilation. Remove these and 
+    test the rest of the exercises.
+*/
+var defaultProjects = allProjects 
+    -  GetFiles(buildDir + "**/DotDsl.csproj")
+    -  GetFiles(buildDir + "**/Hangman.csproj")
+    -  GetFiles(buildDir + "**/React.csproj");
+
+/** These projects have full and working implementations, and have
+    all the tests enabled. These should pass without any changes.
+*/
+var refactoringProjects = GetFiles(buildDir + "**/TreeBuilding.csproj")
+    + GetFiles(buildDir + "**/Ledger.csproj")
+    + GetFiles(buildDir + "**/Markdown.csproj");
+
+
+Task("Clean")
+	.Does(() => {
+		CleanDirectory(buildDir);   
+    });
+
+
+// Copy everything to build so we make no changes in the actual files.
+Task("CopyExercises")
+    .IsDependentOn("Clean")
+    .Does(() => {
+        CopyDirectory(sourceDir, buildDir);
+    });
+
+
+Task("RestoreNugetPackages")
+    .IsDependentOn("CopyExercises")
+    .Does(() => {
+       foreach (var project in projectDirs) {
+           DotNetCoreRestore(project.ToString());
+       }
+    });
+
+
+Task("RestoreAndBuildStubImplementations")
+    .IsDependentOn("RestoreNugetPackages")
+    .Does(() => {
+        foreach (var project in defaultProjects) {
+            DotNetCoreBuild(project.ToString());
+        }
+});
+
+Task("EnableAllTests")
+    .IsDependentOn("RestoreAndBuildStubImplementations")
+    .Does(() => {
+        var testFiles = GetFiles(buildDir + "**/*Test.cs");
+        foreach (var testFile in testFiles) {
+            var skipRE = @"Skip\s*=\s*""Remove to run test""";
+            ReplaceRegexInFiles(buildDir + "**/*Test.cs", skipRE, "");
+        }
+    });
+
+Task("RestoreAndTestRefactoringProjects")
+    .IsDependentOn("EnableAllTests")
+    .Does(() => {
+        foreach (var project in refactoringProjects) {
+            DotNetCoreTest(project.ToString());
+        }
+});
+
+
+Task("ReplaceStubWithExample")
+    .IsDependentOn("RestoreAndTestRefactoringProjects")
+    .Does(() => {
+        foreach (var project in allProjects) {
+            var projectDir = project.GetDirectory();
+            var projectName = project.GetFilenameWithoutExtension();
+            var stub = projectDir.GetFilePath(projectName).AppendExtension("cs");
+            var example = projectDir.GetFilePath("Example.cs");
+            
+            DeleteFile(stub);
+            MoveFile(example, stub);
+        }
+    });
+
+Task("TestUsingExampleImplementation")
+    .IsDependentOn("ReplaceStubWithExample")
+    .Does(() => {
+        foreach (var project in allProjects) {
+            DotNetCoreTest(project.ToString());
+        }
+    });
+
+
+Task("Default")
```

Comment:
I tried this. But `build.ps1` already has set the command line parameter `[string]$Target = "Default"`. Besides the `Default` task is set as the starting tasks in some of the cake build files that I saw on the internet, perhaps this is the convention?